### PR TITLE
feat(repo): Improve error message for missing repositories.yaml

### PR DESCRIPTION
**Pull request by mgalgs**
New users to helm don't always run `helm init` (e.g. if their cluster
already has helm installed).  The user's initial interaction with any of
helm's repository commands (e.g. `helm repo list`) will then be an error
message due to a missing repositories.yaml in their local config.

Give the user a little hint about how to fix the error without them having
to hunt through the man/help pages.

### CODE REVIEW

**Comment by bacongobbler** [comment]
Hey @mgalgs! Would you mind writing out a quick unit test that covers this so we don't accidentally miss this in future refactors? Thanks so much! This is a great fix.

**Comment by mgalgs** [comment]
I'm happy to add a test, but might need some hints :joy: 

I haven't changed any behavior here, just added a print.  I'm new to testing in `go` but I don't see any examples in the existing tests (looking at `repo_test.go` et al.?) where we're parsing print statements?

File: pkg/repo/repo.go [hunk 1]
```diff
@@ -55,6 +56,12 @@ func NewRepoFile() *RepoFile {
 func LoadRepositoriesFile(path string) (*RepoFile, error) {
 	b, err := ioutil.ReadFile(path)
 	if err != nil {
+		if strings.HasSuffix(err.Error(), "no such file or directory") {
```
Comment (bacongobbler) [inline pkg/repo/repo.go]: It'd be better to change this to [`if os.IsNotExist(err) {`](https://golang.org/pkg/os/#IsNotExist)

**Comment by bacongobbler** [comment]
okay well first things I'd do is

a) lemme grab you an example. Essentially what you want to do is test that calling `LoadRepositoriesFile` on a directory that does not exist returns the expected error.

**Comment by bacongobbler** [comment]
Here's an example from Draft: https://github.com/Azure/draft/blob/680c348eea578cac27af4b9b74b89bac6cae0630/pkg/draft/pack/load_test.go#L33-L35

**Comment by mgalgs** [comment]
What I really meant is that this will be testing existing behavior, not my patch per se.  But in any case, I'm happy to add such a test.  Enjoying the learning process :)

File: pkg/repo/repo.go (hunk 1 above)
Comment (mgalgs) [reply pkg/repo/repo.go]: Done

Comment (mgalgs) [reply pkg/repo/repo.go]: Doh wait, I don't think you mean it like that :grimacing: Let me try one more time :joy:

Comment (mgalgs) [reply pkg/repo/repo.go]: All right, think I got it :sweat_smile:

File: pkg/repo/repo.go [hunk 2]
```diff
@@ -55,6 +55,12 @@ func NewRepoFile() *RepoFile {
 func LoadRepositoriesFile(path string) (*RepoFile, error) {
 	b, err := ioutil.ReadFile(path)
 	if err != nil {
+		if os.IsNotExist(err) {
+			fmt.Printf("Couldn't load repositories file (%s).\n"+
```
Comment (bacongobbler) [inline pkg/repo/repo.go]: Oh sorry, NOW I see what you mean. We should be returning `nil, fmt.Errorf()` instead or displaying on stdout. That way clients using `LoadRepositoriesFile` can have control whether or not this error message should be displayed on stdout or not. THEN we can check in pkg/repo/repo_test.go for the error message.

Comment (mgalgs) [reply pkg/repo/repo.go]: Aah ok. Initial attempt is up.

**Comment by mgalgs** [comment]
I could be wrong but the circleci failure seems unrelated.

**Comment by bacongobbler** [comment]
*much* better, thank you.

Yeah that's an odd failure. I've kicked off another build just to make sure. :)

**Comment by bacongobbler** [comment]
The root cause appears to be https://github.com/Masterminds/glide/issues/955. This is because we are living on the bleeding edge of glide in our CI tests. Now that it's been fixed upstream (thank you as always, @mattfarina :heart:), let's restart the job and make sure everything works.

**Review by bacongobbler — APPROVED** [review]

**Pull request merged**